src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -3166,13 +3167,14 @@ NetworkOPsImp::transJson(
 
     std::string const hash = to_string(transaction->getTransactionID());
     MultiApiJson multiObj{jvObj};
-    visit<RPC::apiMinimumSupportedVersion, RPC::apiMaximumValidVersion>(
-        multiObj,  //
-        [&](Json::Value& jvTx, unsigned int apiVersion) {
+    forAllApiVersions(
+        multiObj.visit(),  //
+        [&](Json::Value& jvTx, auto v) {
+            constexpr unsigned version = decltype(v)::value;
             RPC::insertDeliverMax(
-                jvTx[jss::transaction], transaction->getTxnType(), apiVersion);
+                jvTx[jss::transaction], transaction->getTxnType(), version);
 
-            if (apiVersion > 1)
+            if constexpr (version > 1)
```

Comment:
Ugh that would make things pretty complex elsewhere (because we can not longer rely on overloading alone), only to support this ugly syntax `template operator()<version>`

What if I simply replaced the lambda signature at line 3172 with this ?

```cpp
        [&]<unsigned Version>(Json::Value& jvTx, std::integral_constant<unsigned, Version>) {
```